memcache/selector.go

```diff
@@ -152,3 +160,49 @@ func (ss *ServerList) PickServer(key string) (net.Addr, error) {
 
 	return ss.addrs[cs%uint32(len(ss.addrs))], nil
 }
+
+func NewRendezvousSelector() RendezvousSelector {
+	return RendezvousSelector{new(ServerList)}
+}
+
+// RendezvousSelector is a ServerSelector which uses the rendezvous hashing algorithm to select a server for a given key.
+// See https://en.wikipedia.org/wiki/Rendezvous_hashing
+type RendezvousSelector struct {
+	*ServerList
+}
+
+// PickServer uses teh rendezvous hashing algorith.
+// It loops over each server, hashes key+server together, and selects the server with the largest hash value.
+// Key -> server mappings should be deterministic and uniformly distributed.
+// This method is threadsafe.
+func (rs RendezvousSelector) PickServer(key string) (net.Addr, error) {
+	buf := keyBufPool.Get().(*[]byte)
+	defer keyBufPool.Put(buf)
```

Comment:
are you sure you need a buffer pool for this? feels like a premature optimization unless you have benchmarked the difference and its big?

Reply:
This just avoids (or I guess amortizes) allocations. The naive checksum+mod implementation was already using this buffer pool, so I thought it was easy enough to add. 

```
BenchmarkRendezvousSelector_PickServer-POOLED   	 2803006	       433.2 ns/op	       0 B/op	       0 allocs/op

BenchmarkRendezvousSelector_PickServer   	 3908797	       308.0 ns/op	      16 B/op	       1 allocs/op
```

I don't have strong opinions!

Reply:
ah i see. it's fine